python/lsst/ts/m2gui/utility_monitor.py

```diff
@@ -0,0 +1,436 @@
+
+    def get_temperature_sensors(self, temperature_group):
+        """Get the temperature sensors in a specific group.
+
+        Parameters
+        ----------
+        temperature_group : enum `TemperatureGroup`
+            Temperature group.
+
+        Returns
+        -------
+        `list [str]`
+            List of the sensors.
+        """
+
+        sensors = list()
+        for sensor in list(self.temperatures.keys()):
+            if sensor.startswith(temperature_group.name):
+                sensors.append(sensor)
+
+        return sensors
+
+    def _report_displacements(self):
+        """Report the displacements."""
+
+        for direction in DisplacementSensorDirection:
+            sensors = self.get_displacement_sensors(direction)
+            displacements = [self.displacements[sensor] for sensor in sensors]
+
+            self.signal_utility.displacements.emit((direction, displacements))
+
+    def get_displacement_sensors(self, direction):
+        """Get the displacement sensors in a specific direction.
+
+        Parameters
+        ----------
+        direction : enum `DisplacementSensorDirection`
+            Direction of the displacement sensors.
+
+        Returns
+        -------
+        `list [str]`
+            List of the sensors.
+        """
+
+        sensors = []
+        for sensor in self.displacements.keys():
+            result = re.match(rf"\AA\d-{direction.name}_z", sensor)
+            if result is not None:
+                sensors.append(result.group())
+
+        return sensors
+
+    def update_power(self, power_type, new_voltage, new_current):
+        """Update the power data.
+
+        Parameters
+        ----------
+        power_type : enum `PowerType`
+            Power type.
+        new_voltage : `float`
+            New voltage value in volt.
+        new_current : `float`
+            New current value in ampere.
+
+        Raises
+        ------
+        `ValueError`
+            Not supported power type.
+        """
+
+        if power_type == PowerType.Motor:
+            power = self.power_motor
+            signal_name = "power_motor"
+        elif power_type == PowerType.Communication:
+            power = self.power_communication
+            signal_name = "power_communication"
+        else:
+            raise ValueError(f"Not supported power type: {power_type!r}.")
+
+        tor = self._get_tor()
+        if (
+            abs(power["voltage"] - new_voltage) >= tor
+            or abs(power["current"] - new_current) >= tor
+        ):
+
+            power["voltage"] = round(new_voltage, self.NUM_DIGIT_AFTER_DECIMAL)
+            power["current"] = round(new_current, self.NUM_DIGIT_AFTER_DECIMAL)
+
+            getattr(self.signal_utility, signal_name).emit(
+                (power["voltage"], power["current"])
+            )
+
+    def _get_tor(self, is_displacement_sensor=False):
+        """Get the torrance.
```

Comment:
What is torrance? I cannot find its translation or meaning. Do you mean tolerance?

https://www.collinsdictionary.com/dictionary/english/torrance

Reply:
Fixed in the commit: [2927fa9](https://github.com/lsst-ts/ts_m2gui/pull/7/commits/2927fa94e6b229312b4685c13217911ee3dca4f0). Thanks!